src/Controls/src/SourceGen/XmlnsDefinitionSourceGenerator.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.Maui.Controls.SourceGen;
+
+using static GeneratorHelpers;
+
+[Generator(LanguageNames.CSharp)]
+public class XmlnsDefinitionSourceGenerator : IIncrementalGenerator
+{
+	public void Initialize(IncrementalGeneratorInitializationContext initContext)
+	{
+		// Read the XmlnsDefinition items data file
+		var xmlnsDefinitionDataProvider = initContext.AdditionalTextsProvider
+			.Where(static file => file.Path.EndsWith(".xmlnsdefinitions", StringComparison.OrdinalIgnoreCase))
```

Comment:
and thus read properties directly from 'normal' msbuild files, and not specially named ones